dm/dm/master/scheduler/scheduler.go

```diff
@@ -1084,6 +1084,46 @@ func (s *Scheduler) GetSubTaskCfgsByTask(task string) map[string]*config.SubTask
 	return cloneM
 }
 
+func (s *Scheduler) GetSubTaskCfgsByTaskAndSource(taskName string, sources []string) (map[string]map[string]config.SubTaskConfig, error) {
+	var ret map[string]map[string]config.SubTaskConfig // task-name->sourceID->*config.SubTaskConfig
+	if len(taskName) == 0 {
+		ret = s.GetSubTaskCfgs()
```

Comment:
could you please explain this if's purpose ? why caller will input a empty taskName?

Reply:
When users would like to get all tasks of some sources, the `taskName` would be empty.